browser/lib/util/webstorage.js

```diff
@@ -9,25 +10,26 @@ var WebStorage = (function() {
 	 * somewhat roundabout way. (If unsupported or no window object,
 	 * will throw on accessing a property of undefined) */
 	try {
-		window.sessionStorage.setItem(test, test);
-		window.sessionStorage.removeItem(test);
+		globalObject.sessionStorage.setItem(test, test);
+        globalObject.sessionStorage.removeItem(test);
 		sessionSupported = true;
 	} catch(e) {
 		sessionSupported = false;
 	}
 
 	try {
-		window.localStorage.setItem(test, test);
-		window.localStorage.removeItem(test);
+        globalObject.localStorage.setItem(test, test);
+        globalObject.localStorage.removeItem(test);
 		localSupported = true;
 	} catch(e) {
 		localSupported = false;
 	}
 
-	function WebStorage() {}
+
+    function WebStorage() {}
 
 	function storageInterface(session) {
-		return session ? window.sessionStorage : window.localStorage;
+		return session ? globalObject.sessionStorage : globalObject.localStorage;
```

Comment:
>  it is way quicker and simpler to just include LocalStorage and be done with

Sorry -- I'm not sure if I'm missing something, but I'm not sure I see the difficulty. Here's a version of WebStorage tweaked to use application-settings based on the nativescript docs; please test and let me know if it works.

(To expand on the reasoning above -- as well as the double abstraction layer objection, Paddy's gone to some lengths to avoid modifying global objects in the lib (see eg most of utils.js), so I'd prefer not to use the lib that monkeypatches localstorage onto the global object. So I'd prefer this solution).

```javascript
var WebStorage = (function() {
	var appSettings = require("application-settings");
		test = 'ablyjs-storage-test';

	function WebStorage() {}

	function set(name, value, ttl) {
		var wrappedValue = {value: value};
		if(ttl) {
			wrappedValue.expires = Utils.now() + ttl;
		}
		return appSettings.setString(name, JSON.stringify(wrappedValue));
	}

	function get(name) {
		var rawItem = appSettings.getString(name);
		if(!rawItem) return null;
		var wrappedValue = JSON.parse(rawItem);
		if(wrappedValue.expires && (wrappedValue.expires < Utils.now())) {
			appSettings.remove(name);
			return null;
		}
		return wrappedValue.value;
	}

	WebStorage.set    = function(name, value, ttl) { return set(name, value, ttl); };
	WebStorage.get    = function(name) { return get(name); };
	WebStorage.remove = function(name) { return appSettings.remove(name); };

	return WebStorage;
})();
```